storybookjs/storybook
# NextJS: Fix Image Context re-use via singleton

## What I did

#23881 was supposed to fix Image Context reuse, but it turns out that `code/frameworks/nextjs/src/preview.tsx` uses the CommonJS export (`code/frameworks/nextjs/dist/image-context.js`), while `code/frameworks/nextjs/src/images/next-image.tsx` etc. uses the ESM export (`code/frameworks/nextjs/dist/image-context.mjs`), causing the Image Context to not be reused and thus parameters specified in `parameters.nextjs.image` and applied in `preview.tsx` won't be available in `next-image.tsx`.

This pull request forces the use of the CommonJS export, even when using `import`, making sure the same Image Context is reused for both ESM and CommonJS imports.

## Checklist for Contributors

### Testing



#### The changes in this PR are covered in the following automated tests:

#### Manual testing

If you check the [Eager story](https://634ff0d0ec053b270775979d-icsyfdvsye.chromatic.com/?path=/story/frameworks-nextjs-image--eager) from the previous pull request you can see the image isn't being eagerly loaded, while the Eager story of this branch properly eager loads the image.

### Documentation



      [MIGRATION.MD](https://github.com/storybookjs/storybook/blob/next/MIGRATION.md)

## Checklist for Maintainers

   <details>
     <summary>Available labels</summary>

     - `bug`: Internal changes that fixes incorrect behavior.
     - `maintenance`: User-facing maintenance tasks.
     - `dependencies`: Upgrading (sometimes downgrading) dependencies.
     - `build`: Internal-facing build tooling & test updates. Will not show up in release changelog.
     - `cleanup`: Minor cleanup style change. Will not show up in release changelog.
     - `documentation`: Documentation **only** changes. Will not show up in release changelog.
     - `feature request`: Introducing a new feature.
     - `BREAKING CHANGE`: Changes that break compatibility in some way with current major version.
     - `other`: Changes that don't fit in the above categories.
   
   </details>

### 🦋 Canary release



This PR does not have a canary release associated. You can request a canary release of this pull request by mentioning the `@storybookjs/core` team here.

_core team members can create a canary release [here](https://github.com/storybookjs/storybook/actions/workflows/canary-release-pr.yml) or locally with `gh workflow run --repo storybookjs/storybook canary-release-pr.yml --field pr=<PR_NUMBER>`_
